### ISSUE TITLE
Cannot restore only the database from a backup of both database and active config.

### ISSUE THREAD

**Issue by izmeez** [backdrop-contrib/backup_migrate#166]
If a manual backup is done of `Database and Active Config Directory` and then a restore of the backup is done selecting only `Database` it fails with the error:

> Unable to restore from file /tmp/backup_migrate_....db_config.tar because a tar file can't be restored to this database.

Restoring both database and active config works but may not be what the user wants to do.

**Comment by argiepiano** [comment]
Yes, this is expected. You can't do a Database restore from a file that contains something else. Perhaps there is lack of clarity in the choice - this choice is not supposed to give you options of what to restore, but is rather a way for the user to indicate the type of backup file you are trying to restore. The file type and the destinations must match. 

BTW this confusion has been pointed out in other issues.

Perhaps there should be a way to prevent the fatal error and instead indicated the mismatch before trying to process the backup restore.

**Comment by izmeez** [comment]
@argiepiano Thank you for the reply. It sounds like it is "By design".

> Perhaps there should be a way to prevent the fatal error and instead indicated the mismatch before trying to process the backup restore.

1. It may make more sense if there is a notice rather than an error.

2. It may require something more to help users understand this restriction: that combined backups can only be used for combined restores. It may affect the workflow of different use cases.

Previously `bam` only did one thing, the database. Now with the option to include the active config with the database the implications of restore is evident.

**Comment by argiepiano** [comment]
I agree this is confusing. I think the way to go here is to check the type of source in a form validation. Also perhaps add some help text to the form.

**Comment by izmeez** [comment]
Now, with a better understanding of the way the combined backup/restore works we have changed our workflow. Some extra help text is a good idea.

**Comment by izmeez** [comment]
This issue is closely related to https://github.com/backdrop-contrib/backup_migrate/issues/93 and they both could use a solution.

**Comment by izmeez** [comment]
I am actively developing and testing a solution for this issue and hope to be able to offer a PR soon that may be suitable for the next release of the module.

**Comment by izmeez** [comment]
I have added a PR to fix this issue that has been "Assisted-by: LLM agent".

The "Database and Active Config Directory" restore source always restored both parts unconditionally, with no way to pick just one. This PR adds that choice, plus two supporting fixes needed to make the choice actually reachable and usable.

The restore form now shows checkboxes - Database, Configuration - when restoring a db_config backup, instead of always restoring both. Checkboxes rather than radios, so any combination (including both) is just a matter of checking the relevant boxes, with no separate "both" option needed.

The "Restore to" dropdown previously defaulted to "Database" regardless of what the selected backup actually contained. It now reads the source recorded on the file at backup time and selects the matching destination automatically. Without this, most users would never notice the new checkboxes above, since they'd need to first manually change a dropdown that gave no indication it was wrong.

When testing with more than 5 saved backups revealed no pager because the pager code was not working. This is fixed in the PR.

During development of the PR other issues were identified and will be added as followup issues:

1. The `Restore` link on `Saved Backups` and `Destinations` uses a different path and should be redirected to this new approach on the `Restore` page.
2. Entire site backups require work to provide restore options for the types of files included in the backup.
